File: download/src/main/java/com/huison/download/DownloadCallback.java
package com.huison.download;

/**
 * Created by huisonma on 2019/5/15.
 */
public interface DownloadCallback {

    void onUpdate(String url, int progress, long downloadedLength, long totalLength);

    void onPause(String url, long downloadedLength, long totalLength);

    void onSuccess(String url, String filePath);

    void onFailed(String url, Exception e);
}

File: ktdownload/src/main/java/com/huison/kotlin/download/DownloadCallback.kt
package com.huison.kotlin.download

import java.lang.Exception

/**
 * Created by huisonma 2019/9/3.
 */
interface DownloadCallback {

    fun onUpdate(url: String, progress: Int, downloadedLength: Long, totalLength: Long)

    fun onPause(url: String, downloadedLength: Long, totalLength: Long)

    fun onSuccess(url: String, filePath: String)

    fun onFailed(url: String?, e: Exception)
}